nrclient/proxy.go

```diff
@@ -0,0 +1,214 @@
+package nrclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/newrelic-fluent-bit-output/config"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func buildHttpTransport(cfg config.ProxyConfig, nrUrl string) (*http.Transport, error) {
+	proxyResolver, err := getProxyResolver(cfg.IgnoreSystemProxy, cfg.Proxy)
+	if err != nil {
+		return nil, err
+	}
+
+	caCertPool, err := getCertPool(cfg.CABundleFile, cfg.CABundleDir)
+	if err != nil {
+		return nil, err
+	}
+
+	var tlsConfig *tls.Config
+	if cfg.CABundleFile != "" || cfg.CABundleDir != "" {
+		tlsConfig = &tls.Config{RootCAs: caCertPool}
+	}
+
+	transport := &http.Transport{
+		Proxy:                 proxyResolver,
+		DialContext:           (&net.Dialer{KeepAlive: 600 * time.Second}).DialContext,
```

Comment:
should these values be configurable? Maybe comments about why they were chosen, too. (Ignore if this code was copy/pasted from the agent)

Reply:
Same as above